framework/src/media/FocusManager.cpp

```diff
@@ -0,0 +1,128 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <media/FocusManager.h>
+
+namespace media {
+FocusRequest::FocusRequest(int id, std::shared_ptr<FocusChangeListener> listener) : mId(id), mListener(listener)
+{
+}
+
+int FocusRequest::getId()
+{
+	return mId;
+}
+
+std::shared_ptr<FocusChangeListener> FocusRequest::getListener()
+{
+	return mListener;
+}
+
+FocusRequestBuilder::FocusRequestBuilder() : mId(getpid()), mListener(nullptr)
+{
+}
+
+FocusRequestBuilder &FocusRequestBuilder::setFocusChangeListener(std::shared_ptr<FocusChangeListener> listener)
+{
+	mListener = listener;
+	return *this;
+}
+
+FocusRequestBuilder &FocusRequestBuilder::setId(int id)
+{
+	mId = id;
+	return *this;
+}
+
+std::shared_ptr<FocusRequest> FocusRequestBuilder::build()
+{
+	auto focusRequest = std::make_shared<FocusRequest>(mId, mListener);
+	return focusRequest;
+}
+
+FocusManager::FocusRequester::FocusRequester(int id, std::shared_ptr<FocusChangeListener> listener) : mId(id), mListener(listener)
+{
+}
+
+int FocusManager::FocusRequester::getId()
+{
+	return mId;
+}
+
+void FocusManager::FocusRequester::notify(int focusChange)
```

Comment:
If any function do something in only one condition, you'd better check it caller.
```
fun A() {
    run();
}
fun run() {
   if (isCar) {
       drive();
   }
}
```
if isCar is false, flow is A -> Run -> check isCar
```
fun A() {
    if (isCar) {
        run();
    }
}
fun run() {
    drive();
}
```
This case flow is A -> check isCar

Reply:
I think
```
fun A() {
    run();
}
fun run() {
   if (isCar) {
       drive();
   }
}
```
is better, because of preventing mistake

```
fun A() {
    if (isCar) {
        run();
    }
}

fun B() {
    run(); // It can make ASSERT
}

fun run() {
    drive();
}
```

And, `notify` logic will be complicated. So I think current logic is better.